options/options.go

```diff
@@ -144,8 +144,8 @@ func (opts Options) CloneOptions() (*git.CloneOptions, error) {
 
 	var auth transport.AuthMethod
 
-	if strings.HasPrefix(opts.RepoURL, "git") {
-		// using git protocol so needs ssh auth
+	if strings.HasPrefix(opts.RepoURL, "ssh://") || (!strings.Contains(opts.RepoURL, "://") && strings.Contains(opts.RepoURL, ":")) {
```

Comment:
@Tomy2e can we augment this to support using sshauth when `opts.RepoURL` is prefixed with `git`. The reason why `git` is hardcoded is that the default URL github/gitlab provide is `git@github/gitlab...`.

Reply:
Hi @zricethezav! This code already supports URLs provided by GitHub/GitLab. They provide an "scp-like" URL, which is detected by making sure the URL doesn't contain the substring `://` and it contains the `:` character. The `git` prefix you are talking about is the user.

I think it's a simple solution, maybe not the best. The go-git library [uses a regex pattern](https://github.com/go-git/go-git/blob/bf3471db54b0255ab5b159005069f37528a151b7/internal/url/url.go#L20) to detect these URLs.

Basically, it will work whether the user is `git` or something else. SSH URLs are documented [here](https://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols#_the_ssh_protocol) if you need more information. (In the current implementation, it won't work if no user is specified, SSHAuth() will return an error.)

The following command works for me, let me know if  you have issues:
```console
$ go run main.go --repo-url=git@github.com:zricethezav/gitleaks.git
```

Reply:
@Tomy2e perfect, thank you for explaining that 😄